src/inmanta/module.py

```diff
@@ -2283,6 +2480,38 @@ def get_all_python_requirements_as_list(self) -> List[str]:
     def get_module_requirements(self) -> List[str]:
         return [*self.metadata.requires, *(str(req) for req in self.get_module_v2_requirements())]
 
+    def add_module_requirement(self, requirement: InmantaModuleRequirement, add_as_v1_module: bool) -> None:
+        requirements_txt_file_path = os.path.join(self._path, "requirements.txt")
+        if add_as_v1_module:
+            # Add requirement to module.yml file
+            YamlMetadataFileWithRequiresField.add_module_requirement_and_write(self, requirement)
+            # Refresh in-memory metadata
+            with open(self.get_metadata_file_path(), "r", encoding="utf-8") as fd:
+                self._metadata = ModuleV1Metadata.parse(fd)
```

Comment:
The _metadata is used in the constructor to set up other fields, this renders the object inconsistent. 

Could we just document that this operation makes the object invalid or outdated?

Reply:
With the current implementation I don't really see a problem, because this method only updated the requirements in the metadata file. No state is derived from these requirements as far as I can see. But I updated the documentation with a warning message.